### ISSUE TITLE
Bug Report: Unused Parameter in flash_attn Function

### ISSUE THREAD

**Issue by 1120172175** [lucidrains/alphafold3-pytorch#57]
## Description
I have found an issue in the flash_attn function in the alphafold3-pytorch repository. It appears that the mask parameter, which is passed to the function, is not being used within the function. The specific location of the code is in alphafold3_pytorch/attention.py at line 315.

```python
def flash_attn(
        self,
        q: Float['b h i d'],
        k: Float['b h j d'],
        v: Float['b h j d'],
        mask: Bool['b j'] | None = None
    ) -> Float['b h i d']:

        _, heads, seq_len, _ = q.shape

        attn_mask = None

        if exists(mask):
            mask = repeat(mask, 'b j -> b h i j', h = heads, i = seq_len)

        with torch.backends.cuda.sdp_kernel(**self.attn_config._asdict()):
            out = F.scaled_dot_product_attention(
                q, k, v,
                attn_mask = attn_mask,
                scale = self.scale,
                dropout_p = self.dropout if self.training else 0.
            )

        return out
```

## Problem
The mask parameter seems to be modified but not actually used in the scaled_dot_product_attention function call. The attn_mask variable is always set to None, and the modified mask is never assigned to attn_mask.

## Questions

1.	Is this an oversight, or is the mask parameter intentionally unused?
2.	Should the mask parameter be assigned to attn_mask in the scaled_dot_product_attention function call?

**Comment by lucidrains** [comment]
@1120172175 hey Loren, initially i thought flash attention could be applicable somewhere, but now seeing that all attention is biased (and perhaps that is necessary for the results they achieve), i don't think flash attention can be used at all.

let me remove it to save some complexity
